mesonbuild/backend/ninjabackend.py

```diff
@@ -2387,6 +2387,73 @@ def get_link_whole_args(self, linker, target):
         target_args = self.build_target_link_arguments(linker, target.link_whole_targets)
         return linker.get_link_whole_for(target_args) if len(target_args) else []
 
+    def guess_library_absolute_path(self, libname, search_dirs, prefixes, suffixes):
+        for directory in search_dirs:
+            for suffix in suffixes:
+                for prefix in prefixes:
+                    trial = os.path.join(directory, prefix + libname + '.' + suffix)
+                    if os.path.isfile(trial):
+                        return trial
+
+    def guess_external_link_dependencies(self, linker, target, commands, internal):
+        # This is a work around for missing dependency file generation in the linker
+        # https://sourceware.org/bugzilla/show_bug.cgi?id=22843
```

Comment:
Is there a corresponding bug for `lld`? FreeBSD will finish its migration to `lld` this year, and there are other projects already using it. `lld` might be our best hope for implementing a better future.

Reply:
I might need to rework this comment. Because even fixing the binutils bug, meson could not use that without regressions. Because for libraries in the same project meson does further optimization by generating symbol files. That of course needs dependencies managed by meson by adding ninja dependencies that use an additional check inbetween.